doc/crypto/api/ops/signature.rst

```diff
@@ -463,20 +473,56 @@ EdDSA signature algorithms
 
     PureEdDSA requires an elliptic curve key on a twisted Edwards curve. The following curves are supported:
 
-    *   Edwards25519: the Ed25519 algorithm is computed. The output signature is a 64-byte string: the concatenation of :math:`R` and :math:`S` as defined by :RFC:`8032#5.1.6`.
+    *   Edwards25519: the Ed25519 algorithm is computed. The output signature is a 64-byte string: the concatenation of :math:`R` and :math:`S` as defined by :RFC:`8032#5.1.6`. 
 
-    *   Edwards448: Unless you use the `_with_context` functions, the Ed448 algorithm is computed with an empty string as the context. The output signature is a 114-byte string: the concatenation of :math:`R` and :math:`S` as defined by :RFC:`8032#5.2.6`.
+    *   Edwards448: Unless you use the signature functions that accept a context parameter, such as :code:`psa_sign_message_with_context()` and :code:`psa_verify_message_with_context()`, the Ed448 algorithm is computed with an empty string as the context. The output signature is a 114-byte string: the concatenation of :math:`R` and :math:`S` as defined by :RFC:`8032#5.2.6`.
 
     .. note::
         To sign or verify the pre-computed hash of a message using EdDSA, the HashEdDSA algorithms (`PSA_ALG_ED25519PH` and `PSA_ALG_ED448PH`) can be used.
 
         The signature produced by HashEdDSA is distinct from that produced by PureEdDSA.
 
+    .. note::
+        When signatures on the Edwards 25519 curve were originally defined without domain separation. Later the Ed25519ctx and Ed25519ph variants were defined, both of which accept a context string. However, a signature made with Ed25519ctx and an empty context is distinct from a signature made using the Ed25519.
+        
```

Comment:
In hindsight, I think we would have been better off without defining PSA_ALG_EDDSA at all, and had 5 separate algorithms for Edwards signatures (Ed25519, Ed25519ctx, Ed25519ph, Ed448, and Ed448ph). As it is we will now have four, and hard-to-document habaviour differences - PSA_ALG_SUPPORTS_CONTEXT(PSA_ALG_EDDSA) is indeterminate, as it depends on the key curve.